File: sms-sender-archetype/src/main/resources/archetype-resources/src/main/java/__pluginName__/__pluginName__SmsDescriptor.java
package ${package}.${pluginName};

import se.curity.identityserver.sdk.plugin.descriptor.SmsPluginDescriptor;
import se.curity.identityserver.sdk.service.sms.SmsSender;

public final class ${pluginName}SmsDescriptor implements SmsPluginDescriptor<${pluginName}SmsConfig>
{
    @Override
    public Class<? extends SmsSender> getSmsSenderType()
    {
        return ${pluginName}SmsSender.class;
    }

    @Override
    public String getPluginImplementationType()
    {
        return "${pluginName.toLowerCase()}";
    }

    @Override
    public Class<? extends ${pluginName}SmsConfig> getConfigurationType()
    {
        return ${pluginName}SmsConfig.class;
    }    
}

File: kotlin-sms-sender-archetype/src/main/resources/archetype-resources/src/main/kotlin/__pluginName__/__pluginName__SmsDescriptor.kt
package ${package}.${pluginName}

import se.curity.identityserver.sdk.plugin.descriptor.SmsPluginDescriptor

class ${pluginName}SmsDescriptor: SmsPluginDescriptor<${pluginName}SmsConfig>
{
    override fun getSmsSenderType() = ${pluginName}SmsSender::class.java

    override fun getPluginImplementationType() = "${pluginName.toLowerCase()}"

    override fun getConfigurationType() = ${pluginName}SmsConfig::class.java
}
